Bottom: 15ec088bcc2a7d161a305c58076a862060b8d57c
Top:    87b979fc647e6cfacbbda727f84fabecde4e72a0
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-12 12:06:41 +0800

Refresh of tuntap-introduce-tx-skb-ring

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 91aca2c..e4c981a 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1559,7 +1559,7 @@ static ssize_t tun_do_read(struct tun_struct *tun, struct tun_file *tfile,
 		return -EIO;
 
 	if (tun->flags & IFF_TX_RING) {
-		unsigned head, tail;
+		unsigned long head, tail;
 		struct tun_desc *desc;
 
 		spin_lock(&tfile->rlock);
@@ -1730,6 +1730,7 @@ static int tun_peek_len(struct socket *sock)
 	if (tun->flags & IFF_TX_RING) {
 		unsigned long head = ACCESS_ONCE(tfile->head);
 		unsigned long tail = ACCESS_ONCE(tfile->tail);
+
 		if (head != tail)
 			ret = tfile->tx_descs[tail].len;
 	} else {
